File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        count = {}

        for num in nums:
            count[num] = count.get(num,0)+1
        
        for key,val in count.items():
            if val > 1:
                return True
        
        return False

        # SC -> 

        

File: 217-contains-duplicate/contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {

        Set<Integer> uniques = new HashSet();

        for(int num : nums){
            if(uniques.contains(num)){
                return true;
            }

            uniques.add(num);
        }

        return false;
        
    }
}

// SC -> O(N)
// TC -> O(N)